File: src/aoj/alsd1/golang/2_A_bubble_sort.go
package main

import "fmt"
import "strings"

func scanNums(len int) (nums []int) {
  var num int
  for i := 0; i < len; i++ {
    fmt.Scan(&num)
    nums = append(nums, num)
  }
  return
}

func output(a *[]int) {
  str := ""
  for _, e := range *a {
    str += fmt.Sprintf("%d ", e)
  }
  str = strings.TrimRight(str, " ")
  fmt.Println(str)
}

func main() {
  var n int
  fmt.Scan(&n)
  a := scanNums(n)

  flag := 1
  count := 0
  for flag == 1 {
    flag = 0
    for i := (n - 1); i >= 0; i-- {
      j := i - 1
      if j >= 0 && a[i] < a[j] {
        tmp := a[i]
        a[i] = a[j]
        a[j] = tmp
        flag = 1
        count += 1
      }
    }
    if flag == 0 {
      break
    }
  }
  output(&a)
  fmt.Println(count)
}

File: aoj/alsd1/2_a_bubble_sort.cpp
#include<stdio.h>

int main() {
    int n, A[100];
    scanf("%d",&n);
    for(int i = 0; i < n; i++) {
        scanf("%d",&A[i]);
    }

    int cnt = 0;
    int i = 0;
    bool flag = true;
    while(flag) {
        flag = false;
        for(int j = n-1; j >= i+1; j--) {
            if(A[j] < A[j-1]) {
                int tmp = A[j-1];
                A[j-1] = A[j];
                A[j] = tmp;
                cnt++;
                flag = true;
            }
        }
        i++;
    }

    for(int i = 0; i < n; i++) {
        if(i != n - 1) {
            printf("%d ",A[i]);
        }else{
            printf("%d\n",A[i]);
        }
    }
    printf("%d\n",cnt);
}
